File: src/main/java/com/github/thorben/webnd/domain/character/Attribute.java
package com.github.thorben.webnd.domain.character;

import com.github.thorben.webnd.foundation.AbstractEntity;

import javax.persistence.*;

@Entity(name = "Attribute")
@Table(name = "attributes")
public class Attribute extends AbstractEntity<Integer> {

	@Id
	@GeneratedValue
	private Integer id;
	@MapsId
	@JoinColumn(name = "id")
	private AttributeType type;
	@Column(name = "attribute_value")
	private String value;
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "character_id")
	private Character character;

	protected Attribute() {
		// JPA
	}

	public Attribute(AttributeType type, String value) {
		this.type = type;
		this.value = value;
	}

	public AttributeType getType() {
		return type;
	}

	public String getValue() {
		return value;
	}

	@Override
	public Integer getId() {
		return id;
	}
}

File: src/main/kotlin/com/github/thorben/webnd/domain/character/Attribute.kt
package com.github.thorben.webnd.domain.character

import com.github.thorben.webnd.foundation.AbstractEntity
import javax.persistence.*

@Entity(name = "Attribute")
@Table(name = "attributes")
class Attribute : AbstractEntity<Int?> {
	@Id
	@GeneratedValue
	override val id: Int? = null

	@MapsId
	@JoinColumn(name = "id")
	var type: AttributeType? = null
		private set

	@Column(name = "attribute_value")
	var value: String? = null
		private set

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "character_id")
	private val character: Character? = null

	protected constructor() {
		// JPA
	}

	constructor(type: AttributeType?, value: String?) {
		this.type = type
		this.value = value
	}

}